### ISSUE TITLE
Need a GPU time metric

### ISSUE THREAD

**Issue by jmellorcrummey** [HPCToolkit/hpctoolkit#364]
While looking at data for a large Sandia application, I could see lots of time used by "gpu copyin", "gpu copyout", "gpu sync", and a few kernels. There is no sort order that readily enables one to find and assess how the GPU spends its time.

We need a GPU time (s) metric that is the sum of all GPU time-based metrics.

**Comment by Jokeren** [comment]
@jmellorcrummey what is ` , , , `?

**Comment by jmellorcrummey** [comment]
> @jmellorcrummey what is `, , ,`?

markdown replaced the gpu placeholders with a blank. I removed the less than and greater than so you can see them.

**Comment by Jokeren** [comment]
I see. Yes, we mentioned GPU time in our paper "Tools for top-down performance analysis of GPU-accelerated applications". Section 5.1 introduces `G_i` for each calling context `i`. Previously, we used hpcviewer to sum up *mem copy*, *sync*, and *kernel* times. For user's convenience, we can build this functionality into hpcrun.

**Comment by blue42u** [comment]
We've (me + @jmellorcrummey) previously discussed the possibility of a metric hierarchy. That would add the proper groupings needed to decode GPU metrics easily, e.g.
```
GPU
|`- transfer
|   |`- copyin
|   |`- copyout
|    `- etc...
|`- sync
|    `- etc...
 `- kernel
     `- etc...
```
It requires deep changes to experiment.xml and the Viewer though, so for now it might make sense to add a temporary sum metric into Run.

**Comment by jmellorcrummey** [comment]
for now, we will add it to hpcrun. the metric hierarchy that you described above should also include 

GPU
`- stall
      |`- ifetch
      |`- memory
      ...

**Comment by jmellorcrummey** [comment]
fixed with PR #368
